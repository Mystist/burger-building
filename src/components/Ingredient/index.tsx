import { baseUrl } from "../../api";
import { IngredientsType } from "../../types";

export default function Ingredient({item, onClick}: { item: IngredientsType, onClick: () => void }) {
  return (<>
    <div key={item.id} className="relative">
      <div className="relative w-full overflow-hidden rounded-lg">
        <button className="w-full" onClick={onClick}>
          <img src={`${baseUrl}/img/${item.src}`} alt="ingredient" className="h-[70px] w-full object-contain object-center" />
        </button>
      </div>
      <div className="relative mt-4 text-center">
        <h3 className="text-sm font-medium text-gray-900">{item.name}</h3>
      </div>
    </div>
  </>)
}