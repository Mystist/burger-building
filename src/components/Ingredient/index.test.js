import renderer from 'react-test-renderer'
import Ingredient from './'

test('Should render component', () => {
  const item = {'name': 'egg', 'src': 'egg.png'}

  const component = renderer.create(
    <Ingredient item={item} onClick={() => {}} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
