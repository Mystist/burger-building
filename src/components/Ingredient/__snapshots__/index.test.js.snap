// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<div
  className="relative"
>
  <div
    className="relative w-full overflow-hidden rounded-lg"
  >
    <button
      className="w-full"
      onClick={[Function]}
    >
      <img
        alt="ingredient"
        className="h-[70px] w-full object-contain object-center"
        src="https://xm-crm-react-exercise-server.herokuapp.com/img/egg.png"
      />
    </button>
  </div>
  <div
    className="relative mt-4 text-center"
  >
    <h3
      className="text-sm font-medium text-gray-900"
    >
      egg
    </h3>
  </div>
</div>
`;
